### PR TITLE
add type converter for batch write

### PR DESCRIPTION
**Pull request by marsishandsome**
currently only support convert to:
1. TINYINT 
2. SMALLINT 
3. MEDIUMINT
4.  INT 
5. BIGINT
6. Double 
7. Float

and only support convert from:
1. BooleanType
2. ByteType
3. ShortType
4. IntegerType
5. LongType
6. FloatType
7. DoubleType
8. StringType

other convert type will provide in following PRs

### CODE REVIEW

**Comment by marsishandsome** [comment]
@birdstorm @zhexuany PTAL

**Comment by marsishandsome** [comment]
@birdstorm @zhexuany PTAL

File: core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+          s"do not support converting from ${value.getClass} to column type: ${targetColumnInfo.getType}"
+        )
+    }
+
+    targetColumnInfo.getType.getType match {
+      case MySQLType.TypeFloat =>
+        val lowerBound = -java.lang.Float.MAX_VALUE
+        val upperBound = java.lang.Float.MAX_VALUE
+
+        if (java.lang.Double.compare(result, upperBound) > 0) {
```
Comment (birdstorm) [inline core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala]: I doubt whether we should use Double.compare to compare double values. It should be better to use EPSILON in order to reduce the impact of precision loss.

Comment (marsishandsome) [reply core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala]: fix

File: core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    Math.round(v).longValue()
+
+  private def doubleToLong(v: java.lang.Double): java.lang.Long =
+    Math.round(v)
+
+  private def stringToDouble(v: String): java.lang.Double =
+    java.lang.Double.parseDouble(v)
+
+  private def stringToLong(v: String): java.lang.Long =
+    doubleToLong(stringToDouble(v))
```
Comment (zhexuany) [inline core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala]: why not Longs.tryParse?

**Comment by zhexuany** [comment]
except for one comment, other LGTM

File: core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala (hunk 2 above)
Comment (marsishandsome) [reply core/src/main/scala/com/pingcap/tispark/utils/TiConverter.scala]: fix

**Pull request merged**